src/main/java/de/tk/opensource/privacyproxy/util/ProxyHelper.java

```diff
@@ -17,12 +22,48 @@ public class ProxyHelper {
     public static final int ROUTING_TIMEOUT_MILLISECONDS = 5000;
     private static final Logger LOGGER = LoggerFactory.getLogger(ProxyHelper.class);
 
-    private final Proxy proxy;
+    private final String proxyHost;
+    private final Integer proxyPort;
     private final String nonProxyHosts;
+    private final HttpHost httpProxyHost;
 
-    public ProxyHelper(Proxy proxy, String nonProxyHosts) {
-        this.proxy = proxy;
+    private Proxy proxy;
+    private DefaultRoutePlanner proxyRoutePlanner;
+
+    public ProxyHelper(Proxy proxy, final String proxyHost, final Integer proxyPort, final String nonProxyHosts) {
+        this.proxyHost = proxyHost;
+        this.proxyPort = proxyPort;
         this.nonProxyHosts = nonProxyHosts;
+        this.proxy = proxy;
+        this.httpProxyHost = (this.proxyHost != null && this.proxyPort != null) ? new HttpHost(this.proxyHost, this.proxyPort) : null;
+    }
+
+    private Proxy getProxy() {
```

Comment:
Was hältst du davon dem Proxy-Helper einen zweiten Konstruktor zu spendieren und die Logik (bzw. dann nur noch die Zuweisung oder Instanziierung des Proxy-Objektes) aus dem Getter in den jeweiligen Konstruktor zu verlagern?